File: solutions/csharp/1207-unique-number-of-occurrences.cs
public class Solution {
    public bool UniqueOccurrences(int[] arr) {
        Dictionary<int, int> count = new Dictionary<int, int>();
        foreach (int num in arr) {
            if (count.ContainsKey(num)) {
                count[num]++;
            } else {
                count[num] = 1;
            }
        }
        HashSet<int> uniqueCount = new HashSet<int>(count.Values);
        return uniqueCount.Count == count.Count;
    }
}

File: solutions/javascript/1207-unique-number-of-occurrences.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function(arr) {
    let count = {};
    for (let num of arr) {
        if (num in count) {
            count[num]++;
        } else {
            count[num] = 1;
        }
    }
    let uniqueCount = new Set(Object.values(count));
    return uniqueCount.size === Object.keys(count).length;    
};
